polaris/benchmark/_base.py

```diff
@@ -103,7 +103,7 @@ class BenchmarkSpecification(BaseArtifactModel, ChecksumMixin):
     target_cols: ColumnsType
     input_cols: ColumnsType
     split: SplitType
-    metrics: Union[str, Metric, list[str | Metric]]
+    metrics: str | Metric | GroupedMetric | list[str | Metric | GroupedMetric]
```

Comment:
**Comment**
This might be out of scope for the PR, but the type here should really be `set[Metric | GroupedMetric]`.


The `_validate_metrics` validator below should made into a **before** validator that normalizes the input.